src/web/client/utilities/urlBuilderUtil.ts

```diff
@@ -227,4 +227,42 @@ export function isWebFileWithLazyLoad(fsPath: string): boolean {
         fsPath.includes(WebExtensionContext.rootDirectory.fsPath) &&
         fsPath.includes(SCHEMA_WEBFILE_FOLDER_NAME) &&
         !isPreloadedContent;
-}
\ No newline at end of file
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export async function getOrCreateSharedWorkspace(config: any) {
+    const getWorkspaceResponse = await fetch(
+        `${config.dataverseOrgUrl}/api/data/v9.2/sharedworkspaces`,
+        {
+            headers: config.headers,
+            method: "GET",
+        }
+    );
+
+    const getWorkspaceResult = await getWorkspaceResponse.json();
+
+    if (getWorkspaceResult.value.length) {
+        for (const workspace of await getWorkspaceResult.value) {
+            if (workspace.name === `Site-${WebExtensionContext.currentSchemaVersion === "PortalSchemaV1" ? 'v1' : 'v2'}-${config.websiteid}`) {
+                return workspace;
+            }
+        }
+    }
+
+    const createWorkspaceResponse = await fetch(
+        `${config.dataverseOrgUrl}/api/data/v9.2/sharedworkspaces`,
+        {
+            headers: {
+                ...config.headers,
+                Prefer: "return=representation",
+            },
+            method: "POST",
+            body: JSON.stringify({
+                name: config.websiteid,
```

Comment:
Critical - this needs to follow same pattern as the studio container name - else we will have complete mis-match in read write data

Reply:
addressed in latest commit